### PR TITLE
Enable support for ActiveSupport 7

### PR DESCRIPTION
**Pull request by landongrindheim**
We encountered an issue recently where upgrading to ActiveSupport 7 caused our builds to fail. Upon investigation, I found that `byebug` was no longer a dependency of ActiveSupport. The library now depends on `debug`, which is a gem now but will ship with Ruby 3.1.

### CODE REVIEW

File: common/lib/dependabot/notifications.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # frozen_string_literal: true
 
+require "active_support"
```
Comment (landongrindheim) [inline common/lib/dependabot/notifications.rb:3]: Per a [comment](https://github.com/rails/rails/issues/43889#issuecomment-995364395) on rails/rails, 

> Requiring individual files from Active Support without first requiring the top-level active_support is not supported.

This played out in the test suite for the previous commit, with errors of the following type:

```plaintext
      Failure/Error:
        Dependabot.instrument(
          Notifications::FILE_PARSER_PACKAGE_MANAGER_VERSION_PARSED,
          ecosystem: "bundler",
          package_managers: {
            "bundler" => version
          }
        )

      NameError:
        uninitialized constant ActiveSupport::IsolatedExecutionState
```

**Review by jurre — APPROVED** [review]
Thanks!

File: common/lib/dependabot/notifications.rb (hunk 1 above)
Comment (jurre) [reply common/lib/dependabot/notifications.rb:3]: Nice find!

**Comment by landongrindheim** [comment]
For posterity's sake: The switch from `byebug` to `debug` wasn't necessary in every case here, but we otherwise would be maintaining two dependencies.

File: common/lib/dependabot/shared_helpers.rb [hunk 2]
```diff
@@ -87,7 +87,7 @@ def self.run_helper_subprocess(command:, function:, args:, env: nil,
       if ENV["DEBUG_FUNCTION"] == function
         puts helper_subprocess_bash_command(stdin_data: stdin_data, command: cmd, env: env)
         # Pause execution so we can run helpers inside the temporary directory
-        byebug # rubocop:disable Lint/Debugger
+        binding.break # rubocop:disable Lint/Debugger
```
Comment (jurre) [inline common/lib/dependabot/shared_helpers.rb:90]: Could have used the generic `debugger` here which is aliased in most (all?) debuggers, but 🤷

Comment (landongrindheim) [reply common/lib/dependabot/shared_helpers.rb:90]: Today I learned!

**Pull request merged**